dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04200CreateDefaultVanityURL.java

```diff
@@ -0,0 +1,241 @@
+            dc.addParam(field.get("fieldName"));
+            dc.addParam(field.get("fieldType"));
+            dc.addParam(field.get("fieldRelationType"));
+            dc.addParam(field.get("fieldContentlet"));
+            dc.addParam(field.get("isRequired"));
+            dc.addParam(field.get("isIndexed"));
+            dc.addParam(field.get("isListed"));
+            dc.addParam(field.get("velocityVarName"));
+            dc.addParam(field.get("sortOrder"));
+            dc.addParam(field.get("values"));
+            dc.addParam(field.get("regexCheck"));
+            dc.addParam(field.get("hint"));
+            dc.addParam(field.get("defaultValue"));
+            dc.addParam(field.get("isFixed"));
+            dc.addParam(field.get("isReadOnly"));
+            dc.addParam(field.get("isSearchable"));
+            dc.addParam(field.get("isUnique"));
+            dc.addParam(new Date());
+            dc.loadResult();
+
+            i++;
+        }
+
+
+    }
+
+    /**
+     * This method creates each field that is required for the Vanity URL Content Type and return a
+     * list with them.
+     *
+     * @return list of fields
+     */
+    private List<Map<String, Object>> setBaseVanityURLFields() {
+        List fields = new ArrayList();
+
+        fields.add(setFieldElements("Order", TEXT_FIELD, "",
+                "integer2", true, true, false, "order", INT_6, "", "", "", "0", true, false, true));
+
+        fields.add(setFieldElements("Action", SELECT_FIELD, "",
+                "integer1", true, true, false, "action", INT_5,
+                "200 - Forward|200\r\n301 - Permanent Redirect|301\r\n302 - Temporary Redirect|302\r\n401 - Auth Required|401\r\n403 - Auth Failed|403\r\n404 - Missing|404\r\n500 - Error|500",
+                "", "", "", true, false, true));
+
+        fields.add(
+                setFieldElements("Forward To", CUSTOM_FIELD, "",
+                        "text4", true, true, true, "forwardTo", INT_4,
+                        "$velutil.mergeTemplate('/static/content/file_browser_field_render.vtl')",
+                        "", "", "", true, false, false));
+
+        fields.add(
+                setFieldElements("Uri", TEXT_FIELD, "", "text3",
+                        true, true, true, "uri", INT_3, "", "", "", "", true, false, true));
+
+        fields.add(setFieldElements("Site", CUSTOM_FIELD, "",
+                "text2", true, true, true, "site", INT_2,
+                "$velutil.mergeTemplate('/static/content/site_selector_field_render.vtl')", "", "",
+                "", true, false, false));
+
+        fields.add(setFieldElements("Title", TEXT_FIELD, "",
+                "text1", true, true, true, "title", INT_1, "", "", "", "", true, false, true));
+
+        return fields;
+    }
+
+    /**
+     * This method sets the value of each element that is needed when creating a Field.
+     */
+    private Map<String, Object> setFieldElements(String fieldName, String fieldType,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 16 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)